pages/catalog/CatalogBookHandler.inc.php

```diff
@@ -196,15 +201,15 @@ function download($args, $request, $view = false) {
 			$inline = false;
 			if (!HookRegistry::call('CatalogBookHandler::download', array(&$this, &$publishedMonograph, &$publicationFormat, &$submissionFile, &$inline))) {
 				import('lib.pkp.classes.file.SubmissionFileManager');
-				$monographFileManager = new SubmissionFileManager($publishedMonograph->getContextId(), $monographId);
+				$monographFileManager = new SubmissionFileManager($publishedMonograph->getContextId(), $publishedMonograph->getId());
 				return $monographFileManager->downloadFile($fileId, $revision, $inline);
 			}
 		}
 
 		// Fall-through: user needs to pay for purchase.
 
 		// Users that are not logged in need to register/login first.
-		if (!$user) return $request->redirect(null, 'login', null, null, array('source' => $request->url(null, null, null, array($monographId, $representationId, $fileIdAndRevision))));
+		if (!$user) return $request->redirect(null, 'login', null, null, array('source' => $request->url(null, null, null, array($monographId, $representationId, $bestFileId))));
```

Comment:
Is there possible ambiguity between `$fileIdAndRevision` (e.g. `123-456`) and public IDs? So far we're disallowing numeric IDs for public IDs, but maybe we also need to disallow two hyphenated numbers.